pkg/plugin/work-in-progress/event_handler_gh_test.go

```diff
@@ -49,6 +50,68 @@ var _ = Describe("Work In Progress Plugin features", func() {
 		HaveTargetURL(fmt.Sprintf("%s/%s/%s.html", docStatusRoot, "failure", wip.InProgressDetailsPageName)),
 	)
 
+	Context("Pull Request label change trigger", func() {
+		BeforeEach(func() {
+			defer gock.OffAll()
+			handler = &wip.GitHubWIPPRHandler{Client: NewDefaultGitHubClient(), BotName: botName}
+		})
+
+		AfterEach(EnsureGockRequestsHaveBeenMatched)
+
+		It("should mark opened PR as work-in-progress when labeled with WIP", func() {
+			// given
+			NonExistingRawGitHubFiles("work-in-progress.yml", "work-in-progress.yaml")
+
+			gock.New("https://api.github.com").
+				Get("/repos/bartoszmajsak/wfswarm-booster-pipeline-test/issues/4/labels").
+				Reply(200).
+				BodyString(`[{"id": 934813958,` +
+					`"url": "https://api.github.com/repos/bartoszmajsak/wfswarm-booster-pipeline-test/labels/work-in-progress",` +
+					`"name": "work-in-progress", "color": "ededed", "default": false}]`)
```

Comment:
this mock (and all other mocks for the labels listing) is not necessary - the information is taken from the PR